astropy/utils/misc.py

```diff
@@ -382,18 +384,43 @@ class JsonCustomEncoder(json.JSONEncoder):
 
     """
     def default(self, obj):
+        from .. import units as u
         import numpy as np
-        if isinstance(obj, (np.ndarray, np.number)):
+        if isinstance(obj, np.number):
+            return obj.tolist()
+        elif isinstance(obj, u.Quantity):
+            return dict(value=obj.value, unit=obj.unit.to_string())
+        elif isinstance(obj, np.ndarray):  # Must come after Quantity
             return obj.tolist()
         elif isinstance(obj, (complex, np.complex)):
             return [obj.real, obj.imag]
         elif isinstance(obj, set):
             return list(obj)
         elif isinstance(obj, bytes):  # pragma: py3
             return obj.decode()
+        elif isinstance(obj, u.UnitBase):
+            return obj.name
+        elif obj is u.dimensionless_unscaled:
+            obj = 'dimensionless_unit'
         return json.JSONEncoder.default(self, obj)
 
 
+def jsondumps(obj, **kwargs):
```

Comment:
I would say not.  This just obscures the calling code (effectively hiding the astropy `JsonCustomEncoder`).